lib/Conversion/FIRRTLToHW/LowerToHW.cpp

```diff
@@ -1155,12 +1155,126 @@ FIRRTLModuleLowering::lowerExtModule(FExtModuleOp oldModule,
+
+    // Create output file attribute
+    auto outputFileAttr = hw::OutputFileAttr::getFromFilename(
+        builder.getContext(), outputFile.getValue());
+
+    // Create emit.file operations for additional files, deduplicating by
+    // filename
+    SmallVector<Attribute> additionalFiles;
+    llvm::StringSet<> seenFiles;
+
+    // Add the primary file to the seen set
+    auto primaryFileDict = cast<DictionaryAttr>(filesAttr[0]);
+    auto primaryFileName = primaryFileDict.getAs<StringAttr>("name");
+    if (primaryFileName) {
+      seenFiles.insert(primaryFileName.getValue());
+    }
+
+    // Create emit.file operations for additional files (these are usually
+    // additional collateral such as headers or DPI files).
+    for (size_t i = 1; i < filesAttr.size(); ++i) {
+      auto fileDict = cast<DictionaryAttr>(filesAttr[i]);
+      auto fileName = fileDict.getAs<StringAttr>("name");
+      auto fileContent = fileDict.getAs<StringAttr>("content");
+      auto fileOutputFile = fileDict.getAs<StringAttr>("output_file");
+
+      if (fileName && fileContent && fileOutputFile &&
+          !seenFiles.contains(fileName.getValue())) {
+        seenFiles.insert(fileName.getValue());
+
+        auto fileSymbolName = "blackbox_" + fileName.getValue().str();
+        auto emitFile = builder.create<emit::FileOp>(
+            oldModule.getLoc(), fileOutputFile.getValue(), fileSymbolName);
+        builder.setInsertionPointToStart(&emitFile.getBodyRegion().front());
+        builder.create<emit::VerbatimOp>(oldModule.getLoc(), fileContent);
+        builder.setInsertionPointAfter(emitFile);
+
+        auto ext = llvm::sys::path::extension(fileName.getValue());
+        bool excludeFromFileList =
+            (ext == ".h" || ext == ".vh" || ext == ".svh");
+        auto outputFileAttr = hw::OutputFileAttr::getFromFilename(
+            builder.getContext(), fileOutputFile.getValue(),
+            excludeFromFileList);
+        emitFile->setAttr("output_file", outputFileAttr);
+
+        // Reference this file in additional_files
+        additionalFiles.push_back(
+            FlatSymbolRefAttr::get(builder.getContext(), fileSymbolName));
+      }
+    }
+
+    // Get module parameters
+    auto parameters = getHWParameters(oldModule, /*ignoreValues=*/true);
+    if (!parameters)
+      parameters = builder.getArrayAttr({});
+
+    auto verbatimModule = builder.create<sv::SVVerbatimModuleOp>(
+        oldModule.getLoc(), nameAttr, TypeAttr::get(moduleType), content,
+        outputFileAttr, /*per_port_attrs=*/nullptr, /*port_locs=*/nullptr,
+        parameters,
+        additionalFiles.empty() ? nullptr
+                                : builder.getArrayAttr(additionalFiles));
+
+    SymbolTable::setSymbolVisibility(
+        verbatimModule, SymbolTable::getSymbolVisibility(oldModule));
+
+    annos.removeAnnotations([](Annotation anno) {
+      return anno.isClass(verbatimBlackBoxAnnoClass);
+    });
```

Comment:
I think this will work
```suggestion
    annos.removeAnnotation(verbatimBlackBoxAnnoClass);
```